File: heimdall.sample-client/src/test/java/com/miqdigital/runner/javasample/KarateTestRunner.java
package com.miqdigital.runner.javasample;

import static org.junit.Assert.assertEquals;

import java.io.IOException;

import javax.mail.MessagingException;

import org.junit.Test;

import com.intuit.karate.KarateOptions;
import com.intuit.karate.Results;
import com.intuit.karate.Runner;
import com.miqdigital.Heimdall;

/**
 * The type Karate test runner.
 */
@KarateOptions
public class KarateTestRunner {


  private final String pathOfPropertyFile = "src/test/resources/properties/runner.properties";
  private final String karateOutputPath = "target/cucumber-reports";

  /**
   * Heimdall reporting.
   *
   * @throws InterruptedException the interrupted exception
   * @throws NoSuchFieldException the no such field exception
   * @throws IllegalAccessException the illegal access exception
   * @throws IOException the io exception
   */
  @Test
  public void heimdallReporting()
      throws InterruptedException, NoSuchFieldException, IllegalAccessException, IOException,
      MessagingException {

    Results results =
        Runner.path("classpath:features/Karate").reportDir(karateOutputPath).tags("@regression")
            .parallel(10);

    final Heimdall heimdall = new Heimdall();
    heimdall.updateStatusInS3AndNotify(pathOfPropertyFile, karateOutputPath);
    assertEquals("There are scenario failures", 0, results.getFailCount());
  }
}

File: heimdall.sample-client/src/test/java/com/miqdigital/runner/kotlinesample/KarateTestRunner.kt
package com.miqdigital.runner.kotlinesample

import com.intuit.karate.KarateOptions
import com.intuit.karate.Results
import com.intuit.karate.Runner
import com.miqdigital.Heimdall
import org.junit.Assert.assertEquals
import org.junit.Test
import java.io.IOException

/**
 * The type Karate test runner.
 */
@KarateOptions(features = ["classpath:features/Karate"], tags = ["@regression"])
class KarateTestRunner {


  private val pathOfPropertyFile = "src/test/resources/properties/runner.properties"
  private val karateOutputPath = "target/cucumber-reports"

  /**
   * Heimdall reporting.
   *
   * @throws InterruptedException   the interrupted exception
   * @throws NoSuchFieldException   the no such field exception
   * @throws IllegalAccessException the illegal access exception
   * @throws IOException            the io exception
   */
  @Test
  @Throws(InterruptedException::class, NoSuchFieldException::class, IllegalAccessException::class, IOException::class)
  fun heimdallReporting() {

    val results: Results = Runner.path(karateOutputPath).tags("@regression").parallel(10)

    val heimdall = Heimdall()
    heimdall.updateStatusInS3AndNotify(pathOfPropertyFile, karateOutputPath)
    assertEquals("There are scenario failures", 0, results.failCount.toLong())
  }
}